packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -0,0 +1,155 @@
+    context.print.success(`Content changes in Lambda layer ${layerName} detected. Layer version increased to ${latestVersion}`);
+    context.print.warning('Note: You need to run "amplify update function" to configure your functions with the latest layer version.');
+    await setNewVersionPermissions(context, layerName, layerState);
+  }
+  await layerState.setNewVersionHash(); // "finialize" the latest layer version
+  const storedParams = layerState.toStoredLayerParameters();
+  const additionalLayerParams = {
+    layerName,
+    build: true,
+    providerContext: {
+      provider,
+      service: ServiceName.LambdaLayer,
+      projectName: context.amplify.getProjectDetails().projectConfig.projectName,
+    },
+  };
+  const layerParameters: LayerParameters = { ...storedParams, ...additionalLayerParams };
+  updateLayerArtifacts(context, layerParameters, { cfnFile: isNewVersion });
+}
+
+async function setNewVersionPermissions(context: any, layerName: string, layerState: LayerMetadata) {
+  const defaultPermissions: PrivateLayer[] = [{ type: Permission.private }];
+  let usePrevPermissions = true;
+  const latestVersion = layerState.getLatestVersion();
+  const latestVersionState = layerState.getVersion(latestVersion);
+  const hasNonDefaultPerms =
+    latestVersionState.isPublic() || latestVersionState.listAccoutAccess().length > 0 || latestVersionState.listOrgAccess().length > 0;
+  const yesFlagSet = _.get(context, ['parameters', 'options', 'yes'], false);
+  if (yesFlagSet) {
+    context.print.warning(`Permissions from previous layer version carried forward to new version by default`);
+  } else if (hasNonDefaultPerms) {
+    usePrevPermissions = (await prompt(prevPermsQuestion(layerName))).usePrevPerms;
+  }
+  if (!usePrevPermissions) {
+    layerState.setPermissionsForVersion(latestVersion, defaultPermissions);
+  }
+}
+
+// hashes all of the layer contents as well as the files in the layer path (CFN, parameters, etc)
+export const hashLayerResource = async (layerPath: string): Promise<string> => {
+  return (await globby(['*'], { cwd: layerPath }))
+    .map(filePath => fs.readFileSync(path.join(layerPath, filePath), 'utf8'))
+    .reduce((acc, it) => acc.update(it), crypto.createHash('sha256'))
+    .update(await hashLayerVersionContents(layerPath))
+    .digest('base64');
+};
+
+// hashes just the content that will be zipped into the layer version.
+// for efficiency, it only hashes package.json files in the node_modules folder of nodejs layers
+export const hashLayerVersionContents = async (layerPath: string): Promise<string> => {
+  const nodePath = path.join(layerPath, 'lib', 'nodejs');
+  const nodeHashOptions = {
+    files: {
+      include: ['package.json'],
+    },
+  };
+  const pyPath = path.join(layerPath, 'lib', 'python');
+  const optPath = path.join(layerPath, 'opt');
+
+  const joinedHashes = (await Promise.all([safeHash(nodePath, nodeHashOptions), safeHash(pyPath), safeHash(optPath)])).join();
+
+  return crypto
```

Comment:
Refactor this to stick to the result of `hashElement` and don't implement different hashing. A proper pre-globbing for all the paths need to be included then passed to hashElement will do the trick.

- glob first for all the files/folders we need the hash for and then pass it to `hashElement`

Reply:
I spent wayyy too long trying to get a proper glob pattern so that hashElement would take care of it automatically but couldn't figure it out and their documentation is crap. There's either some bug in their implementation or weird behavior with how the folder and file globs interact. Anyway, I gave up on that and just did this